src/vue/common/TokensList.vue

```diff
@@ -0,0 +1,167 @@
+<template>
+  <div class="tokens-list">
+    <drawer :is-shown.sync="isDetailsDrawerShown">
+      <template slot="heading">
+        {{ 'token-details.title' | globalize }}
+      </template>
+      <token-details :token="selectedToken" />
+    </drawer>
+    <div class="token-cards">
+      <a
+        class="token-card"
+        v-for="token in tokens"
+        :key="token.code"
+        @click="selectToken(token)"
+      >
+        <div class="token-card__header">
+          <token-logo
+            class="token-card__logo"
+            :logo-key="token.details.logo.key"
+            :token-code="token.code"
+          />
+        </div>
+        <div class="token-card__info">
+          <p class="token-card__code">
+            {{ token.code }}
+          </p>
+          <p class="token-card__name">
+            {{ token.details.name }}
+          </p>
+          <p
+            v-if="getBalance(token).value"
+            class="token-card__balance"
+          >
+            {{
+              'tokens-page.balance-msg'| globalize({ value: getBalance(token) })
+            }}
+          </p>
+          <p
+            v-else
+            class="token-card__balance token-card__no-balance"
+          >
+            {{ 'tokens-page.no-balance-msg' | globalize }}
+          </p>
+        </div>
+      </a>
+    </div>
+  </div>
+</template>
+
+<script>
+import Drawer from '@/vue/common/Drawer'
+import TokenDetails from '@/vue/common/TokenDetails'
+import TokenLogo from '@/vue/common/TokenLogo'
+
+import { mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+
+export default {
+  name: 'tokens-list',
+  components: {
+    Drawer,
+    TokenDetails,
+    TokenLogo,
+  },
+  props: {
+    tokens: { type: Array, default: _ => [] },
+  },
+  data: _ => ({
+    isDetailsDrawerShown: false,
+    selectedToken: null,
+  }),
+  computed: {
+    ...mapGetters({
+      account: vuexTypes.account,
+    }),
+  },
+  methods: {
+    getBalance (token) {
```

Comment:
Cause this method is called for every token in the list separately

Reply:
hmmm, I’ve got an idea. Let’s add a method to the record so we it will accept (probably) accountBalances object and parse the thing for us. The method will assign the result into smth like `balance` object of the innstance.